libraries/AP_ADSB/AP_ADSB.cpp

```diff
@@ -643,7 +642,7 @@ void AP_ADSB::set_callsign(const char* str, const bool append_icao)
     } // for i
 
     if (append_icao) {
-        snprintf(&out_state.cfg.callsign[4], 5, "%04X", unsigned(out_state.cfg.ICAO_id % 0x10000));
+        hal.util->snprintf(&out_state.cfg.callsign[4], 5, "%04X", unsigned(out_state.cfg.ICAO_id % 0x10000));
```

Comment:
What's this change for?

Reply:
One of my includes of stdio.h has resulted in a build error on Linux boards because of an include order dependency caused by the the snprintf in AP_ADSB.cpp no longer being able to access stdio.h.  Having spoken to @peterbarker  his recommendation was that the snprintf in AP_ADSB should be be using the same function from the hal.  I don't understand the specifics as to why though.  So I can either make this change or include the stdio.h explicitly AP_ADSB.cpp.  There is no functional change.

Reply:
Well, of simply including stdio.h solves it then that seems the best solution. You added stdio to AP_DAL, might as well do it here too.

Reply:
I don't think there's anything wrong with changing this to use hal-util->snprintf() so I think we should avoid the back-and-forth of asking @Gone4Dirt to change it again.  From a quick search through the code we seem to use either.  If we were going to consolidate on one of the two I suspect we would consolidate on the hal_util version.